addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -342,6 +342,38 @@ def _import_retrieve_and_fill_partner(self, invoice, name, phone, mail, vat, cou
             if vat and self.env['res.partner']._run_vat_test(vat, country, invoice.partner_id.is_company):
                 invoice.partner_id.vat = vat
 
+    def _import_retrieve_and_fill_partner_bank_details(self, invoice, bank_details):
+        """ Retrieve the bank account, if no matching bank account is found, create it
+        """
+        if invoice.journal_id.type == 'purchase':
```

Comment:
@JulienVR. as discussed please check refunds, I think the wrong partner is getting set

Reply:
Indeed, I think it should actually be 
```py
if invoice.move_type in ('out_refund', 'in_invoice'):
    partner = invoice.partner_id
elif invoice.move_type in ('out_invoice', 'in_refund'):
    partner = self.env.company.partner_id
 ```
 If you have a doubt, you can try to create a move, then "create and edit" the recipient bank:
 * for an out_invoice: the recipient bank's account holder is the invoice.company_id
 * for an out_refund: the recipient bank's account holder is the invoice.partner_id :warning: (since it's the one that will receive the money)
 * for an in_invoice: invoice.partner_id (the company pays the partner, he will receive the money)
 * for an in_refund: invoice.company_id (the company receives the money instead of paying to the partner) :warning: 
 see for example: 
![out_refund](https://github.com/odoo/odoo/assets/60702485/fafe3c34-6ced-4e0c-aee9-81933f196ada)